snakemake/wrangler_by_sample_setup.smk

```diff
@@ -1,179 +1,76 @@
+configfile: 'wrangler_by_sample.yaml'
+output='/opt/analysis'
 
 rule all:
 	input:
-		final_out
+		setup_finished=output+'/setup_finished.txt',
+#		good_samples=output+'/successfully_extracted_samples.txt',
+		output_configfile=output+'/snakemake_params/wrangler_by_sample.yaml'
 
-rule extract_by_arm:
+rule copy_files:
 	input:
-	params:
-		output_dir='/opt/analysis/analysis',
-		wrangler_dir=output,
-		sif_file=config['miptools_sif'],
-		fastq_dir=config['fastq_dir']
-	resources:
-		time_min=240
-	output:
-		output+'/analysis/{sample}/{sample}_mipExtraction/log.txt'
-	shell:
-		'''
-		singularity exec \
-		-B {params.fastq_dir}:/opt/data \
-		-B {params.wrangler_dir}:/opt/analysis \
-		{params.sif_file} \
-		MIPWrangler mipExtractByArm --masterDir {params.output_dir} --sample {wildcards.sample} --overWriteDirs --minCaptureLength=30
-		'''
-if config['downsample_umi_count']<2**32:
-	rule mip_barcode_correction:
-		input:
-			good_samples=expand(output+'/analysis/{sample}/{sample}_mipExtraction/log.txt', sample=all_samples)
-		params:
-			output_dir='/opt/analysis/analysis',
-			wrangler_dir=output,
-			sif_file=config['miptools_sif'],
-			downsample_seed=config['downsample_seed'],
-			downsample_amount=config['downsample_umi_count']
-		resources:
-			mem_mb=config['memory_mb_per_step'],
-			time_min=20
-		output:
-			barcode_corrections_finished=output+'/analysis/{sample}/{sample}_mipBarcodeCorrection/barcodeFilterStats.tab.txt'
-		shell:
-			'''
-			singularity exec \
-			-B {params.wrangler_dir}:/opt/analysis \
-			{params.sif_file} \
-			MIPWrangler mipBarcodeCorrection --masterDir {params.output_dir} \
-			--downSampleAmount {params.downsample_amount} --downSampleSeed \
-			{params.downsample_seed} --overWriteDirs --sample {wildcards.sample}
-			'''
-else:
-	rule mip_barcode_correction:
-		input:
-			good_samples=expand(output+'/analysis/{sample}/{sample}_mipExtraction/log.txt', sample=all_samples)
-		params:
-			output_dir='/opt/analysis/analysis',
-			wrangler_dir=output,
-			sif_file=config['miptools_sif'],
-			downsample_seed=config['downsample_seed'],
-		resources:
-			mem_mb=config['memory_mb_per_step'],
-			time_min=20
-		output:
-			barcode_corrections_finished=output+'/analysis/{sample}/{sample}_mipBarcodeCorrection/barcodeFilterStats.tab.txt'
-		shell:
-			'''
-			singularity exec \
-			-B {params.wrangler_dir}:/opt/analysis \
-			{params.sif_file} \
-			MIPWrangler mipBarcodeCorrection --masterDir {params.output_dir} \
-			--doNotDownSample --downSampleSeed \
-			{params.downsample_seed} --overWriteDirs --sample {wildcards.sample}
-			'''
-
-
-rule correct_for_same_barcode_contam:
-	input:
-		all_corrected=expand(output+'/analysis/{sample}/{sample}_mipBarcodeCorrection/barcodeFilterStats.tab.txt', sample=all_samples)
-	params:
-		output_dir='/opt/analysis/analysis',
-		wrangler_dir=output,
-		sif_file=config['miptools_sif'],
-	resources:
-		mem_mb=40000,
-		time_min=1440,
-		nodes=20
-	threads: 20
```

Comment:
Here's where the copy_files rule got moved to, I see now. It looks like github swapped the file contents of the setup and finish_run steps somehow, and showed the changes that would convert one file into the other as if they had been made line by line in each file (instead of swapping out the entire contents of both files)